skillmap/src/lib/authClient.ts

```diff
@@ -16,13 +16,12 @@ class AuthClient extends pxt.auth.AuthClient {
         // Show a notification?
         return Promise.resolve();
     }
-    protected onUserProfileChanged(): Promise<void> {
-        const state = this.getState();
-        if (state.profile) {
+    protected async onUserProfileChanged(): Promise<void> {
+        const state = await pxt.auth.getUserStateAsync();
+        if (state?.profile) {
```

Comment:
I saw there was a change to multiplayers `firstName` logic, is that something that would need to be handled in skillmap as well (for my 'name on cert' i.e. https://github.com/microsoft/pxt/pull/9367/files#diff-b64bd726c7c8a82e62558b9df9a313c6c5eea061222664a197d9d6dfe3631c33R546)